pkg/service/helper.go

```diff
@@ -76,14 +77,32 @@ func getClusterTag(ctx context.Context, c client.Client) (string, error) {
 	return infra.Status.InfrastructureName, nil
 }
 
-func expandDiskByDiskDiskAttachment(conn *ovirtsdk.Connection, diskAttachment *ovirtsdk.DiskAttachment, bytes int64) error {
+func expandDiskByDiskDiskAttachment(ctx context.Context, conn *ovirtsdk.Connection, diskAttachment *ovirtsdk.DiskAttachment, bytes int64) error {
+	correlationID := fmt.Sprintf("image_transfer_%s", diskAttachment.MustId())
+
 	diskAttachment.MustDisk().SetProvisionedSize(bytes)
-	_, err := conn.SystemService().VmsService().VmService(diskAttachment.MustVm().MustId()).
+	expandDiskRequest := conn.SystemService().VmsService().VmService(diskAttachment.MustVm().MustId()).
 		DiskAttachmentsService().AttachmentService(diskAttachment.MustId()).
-		Update().DiskAttachment(diskAttachment).Send()
+		Update().DiskAttachment(diskAttachment)
+	expandDiskRequest.Query("correlation_id", correlationID)
+	_, err := expandDiskRequest.Send()
+	if err != nil {
+		return fmt.Errorf("failed to expand disk attachment %s, error: %w", diskAttachment.MustId(), err)
+	}
+	disk, err := getDiskFromDiskAttachment(conn, diskAttachment)
+	if err != nil {
+		return fmt.Errorf("failed to expand disk attachment %s, error: %w", diskAttachment.MustId(), err)
+	}
+	if err = waitForDiskStatusOk(ctx, conn, disk.MustId()); err != nil {
```

Comment:
this should be after the job check, it can get stuck if we fail and the disk remains locked

Reply:
Done